driver-sync/src/test/functional/com/mongodb/client/AbstractRetryableWritesTest.java

```diff
@@ -125,23 +133,115 @@ public void shouldPassAllOutcomes() {
         boolean wasException = false;
         try {
             result = helper.getOperationResults(operation);
-        } catch (Exception e) {
+        } catch (RuntimeException e) {
             wasException = true;
+            assertExceptionState(e, outcome.get("result", new BsonNull()), operation.getString("name").getValue());
+        }
+
+        if (outcome.getBoolean("error", BsonBoolean.FALSE).getValue()) {
+            assertEquals("Expected an error to be thrown", outcome.containsKey("error"), wasException);
+        } else {
+            BsonDocument fixedExpectedResult = outcome.getDocument("result", new BsonDocument());
+            assertEquals(fixedExpectedResult, result.getDocument("result", new BsonDocument()));
         }
 
         if (outcome.containsKey("collection")) {
             List<BsonDocument> collectionData = collection.withDocumentClass(BsonDocument.class).find().into(new ArrayList<BsonDocument>());
             assertEquals(outcome.getDocument("collection").getArray("data").getValues(), collectionData);
         }
+    }
 
-        if (outcome.getBoolean("error", BsonBoolean.FALSE).getValue()) {
-            assertEquals(outcome.containsKey("error"), wasException);
+    private void assertExceptionState(final RuntimeException e, final BsonValue expectedResult, final String operationName) {
```

Comment:
Copied from `AbstractUnifiedTest` unfortunately the retryable writes test format isn't quite unified yet.